File: wear/app/src/main/java/com/gearan/watch/util/GearanLog.kt
package com.gearan.watch.util

import android.util.Log

/**
 * Consistent diagnostic logging. `adb logcat | grep GEARAN` shows everything.
 * NEVER log: private keys, session keys, nonces, health data, notification content.
 * SAS codes are user-visible by design and may be logged as MATCHED/MISMATCH only.
 */
object GearanLog {
    const val BLE = "GEARAN_BLE"
    const val PAIR = "GEARAN_PAIR"
    const val CRYPTO = "GEARAN_CRYPTO"
    const val LINK = "GEARAN_LINK"
    const val SYNC = "GEARAN_SYNC"

    fun ble(msg: String) = Log.i(BLE, msg)
    fun bleWarn(msg: String, t: Throwable? = null) = Log.w(BLE, msg, t)
    fun pair(msg: String) = Log.i(PAIR, msg)
    fun pairWarn(msg: String, t: Throwable? = null) = Log.w(PAIR, msg, t)
    fun crypto(msg: String) = Log.i(CRYPTO, msg)
    fun cryptoWarn(msg: String, t: Throwable? = null) = Log.w(CRYPTO, msg, t)
    fun link(msg: String) = Log.i(LINK, msg)
    fun linkWarn(msg: String, t: Throwable? = null) = Log.w(LINK, msg, t)
    fun sync(msg: String) = Log.i(SYNC, msg)
    fun syncWarn(msg: String, t: Throwable? = null) = Log.w(SYNC, msg, t)
}

File: ios/Sources/GearanCore/GearanLog.swift
import Foundation
import os

/// Redacted diagnostic logging. Never logs keys, nonces, payloads, health or
/// notification content. SAS appears only as MATCHED/MISMATCH.
public enum GearanLog {
    private static let ble = OSLog(subsystem: "com.gearan.ios", category: "GEARAN_BLE")
    private static let pair = OSLog(subsystem: "com.gearan.ios", category: "GEARAN_PAIR")
    private static let crypto = OSLog(subsystem: "com.gearan.ios", category: "GEARAN_CRYPTO")
    private static let link = OSLog(subsystem: "com.gearan.ios", category: "GEARAN_LINK")

    public static func ble(_ msg: String) { os_log("%{public}@", log: ble, type: .info, msg) }
    public static func pair(_ msg: String) { os_log("%{public}@", log: pair, type: .info, msg) }
    public static func crypto(_ msg: String) { os_log("%{public}@", log: crypto, type: .info, msg) }
    public static func link(_ msg: String) { os_log("%{public}@", log: link, type: .info, msg) }
}
